### DISCUSSION TITLE
Incorrect times from QDomni when updating from PSP site

### DISCUSSION THREAD

**Discussion by drsteve** [spacepy/spacepy#628] [Support and Q&A] [answered]
This is on behalf of a user who got in touch to ask about the time range for available "omni2" (aka Qin-Denton) data. Their local data store only ran through mid-2020, and the work they were doing required late 2021.

Given that `spacepy.toolbox.update` is down for the time being (cf. #616), pending the RBSP-ECT SOC website coming back up, I had directed them to the files on the [PSP/IS☉IS SOC](https://spp-isois.sr.unh.edu/data_public/omni/) as discussed in [this answer](https://github.com/spacepy/spacepy/discussions/616#discussioncomment-3003388).

Replacing the `omnidata.h5` and `omni2data.h5` files in `~/.spacepy/data` works for me, as illustrated here:
```
(default310) user@machine:~/.spacepy/data$ wget https://spp-isois.sr.unh.edu/data_public/omni/omni2data.h5
--2022-07-13 12:20:56--  https://spp-isois.sr.unh.edu/data_public/omni/omni2data.h5
Resolving spp-isois.sr.unh.edu (spp-isois.sr.unh.edu)... 132.177.243.133, 2606:4100:38a0:243::133
Connecting to spp-isois.sr.unh.edu (spp-isois.sr.unh.edu)|132.177.243.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 132955926 (127M)
Saving to: ‘omni2data.h5’

omni2data.h5                  100%[=================================================>] 126.80M  6.08MB/s    in 28s

2022-07-13 12:21:24 (4.59 MB/s) - ‘omni2data.h5’ saved [132955926/132955926]

(default310) user@machine:~/.spacepy/data$ ipython
Python 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 07:04:59) [GCC 10.3.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import spacepy.omni as om
This unreleased version of SpacePy is not supported by the SpacePy team.

In [2]: om.omnirange()
Out[2]: (datetime.datetime(1963, 11, 27, 14, 0), datetime.datetime(2022, 1, 19, 23, 0))

In [3]: import spacepy.time as spt

In [4]: tt = spt.tickrange('2021-11-01', '2021-12-01', 1/24)

In [5]: import spacepy.empiricals as emp

In [6]: pp = emp.getPlasmaPause(tt)

In [7]: pp
Out[7]:
array([3.862     , 3.862     , 3.862     , 3.862     , 3.862     ,
       3.862     , 3.862     , 3.862     , 3.862     , 3.862     ,
       3.862     , 3.862     , 4.35859998, 4.35859998, 4.35859998 ...
```

However, the user let me know that after grabbing the data files the dates for the data appear to be interpreted incorrectly.
```
>>> om.omnirange()
(datetime.datetime(3932, 11, 27, 14, 0), datetime.datetime(3991, 1, 20, 23, 0))
```

So I can't reproduce, but I would ask:
- What version of SpacePy? How was it obtained?
- Versions of Python, numpy, ... ?
- Does simply removing the omni2data.h5 and re-downloading it fix things?

Any thoughts from the team appreciated.

**Comment by qianwu2** [comment]
Hi,
Here is my setting.
python 3.7.13
spacepy 0.2.1
numpy 1.21.2

**Reply by drsteve** [reply]
If the answer I just dropped works for you, I'd appreciate if you could mark it as an accepted answer! It's always nice to know that we've resolved an issue...

**Comment by drsteve** [comment, marked as answer]
Thanks for the update.
I'm going to guess that your matplotlib is 3.3 or newer too.

At version 0.2.2 of spacepy we stopped using matplotlib for our RDT time conversion, and at version 3.3 matplotlib changed their reference date for time conversions. I think you've found an unlikely version mismatch and a function where we didn't trap the matplotlib change...

The solution here is just to update your spacepy. We're currently on version 0.3 via pip. You could also roll back your matplotlib to 3.2, but just updating spacepy will work and is recommended.

**Comment by qianwu2** [comment]
Hi Spacepy/Spacepy,
I answered the questions.
I also updated the spacepy, now it works.
Best,
Qian

On Wed, Jul 13, 2022 at 4:20 PM Steve Morley ***@***.***>
wrote:

> If the answer I just dropped works for you, I'd appreciate if you could
> mark it as an accepted answer! It's always nice to know that we've resolved
> an issue...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/spacepy/spacepy/discussions/628#discussioncomment-3141424>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIBNSTAK75NBWG5CIODDEFTVT46KNANCNFSM53QBXVIA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Qian Wu
High Altitude Observatory
National Center for Atmospheric Research
P.O.Box 3000
Boulder, Co 80307-3000
(303) 497 2176
(303) 497 2180

**Comment by qianwu2** [comment]
After updated the spacepy, the dates are correct now. 
Many thanks.
